File: test13.py
def noonerize(numbers):
    num_one = str(numbers[0])
    num_two = str(numbers[1])
    
    # Extract first digits
    first_digit_one = num_one[0]
    first_digit_two = num_two[0]

    # Swap first digits
    num_one_spoonerized = first_digit_two + num_one[1:]
    num_two_spoonerized = first_digit_one + num_two[1:]

    spoonerized_diff = abs(int(num_two_spoonerized) - int(num_one_spoonerized))

    return spoonerized_diff

print(noonerize([123, 456]))  # Output: 267
print(noonerize([12, 34]))    # Output: 22
print(noonerize([55, 55]))    # Output: 0
print(noonerize([1, "abc"]))   # Output: TypeError

File: test13.js
function noonerize(numbers) {
    // Check if both items in the array are numbers
    for (let i = 0; i < numbers.length; i++) {
        if (typeof numbers[i] !== 'number') {
            return "invalid array";
        }
    }

    // Ensure there are exactly two elements in the array
    if (numbers.length !== 2) {
        return "invalid array";
    }

    // Spoonerize the numbers and calculate the positive difference
    const num1 = numbers[0].toString();
    const num2 = numbers[1].toString();
    const spoonerizedDiff = Math.abs(parseInt(num2[0] + num1.substring(1)) - parseInt(num1[0] + num2.substring(1)));

    return spoonerizedDiff;
}

// Example usage:
console.log(noonerize([123, 456])); // Output: 267
console.log(noonerize([12, 34]));   // Output: 22
console.log(noonerize([55, 55]));   // Output: 0
console.log(noonerize([1, "abc"]));  // Output: invalid array
